src/nbl/video/CVulkanCommandBuffer.cpp

```diff
@@ -6,400 +6,891 @@
-                
-                // TODO: Use better container when ready for these stack allocated memories.
-                VkAccelerationStructureBuildGeometryInfoKHR vk_buildGeomsInfos[MaxBuildInfoCount] = {};
+        vk_viewports[i].x = pViewports[i].x;
+        vk_viewports[i].y = pViewports[i].y;
+        vk_viewports[i].width = pViewports[i].width;
+        vk_viewports[i].height = pViewports[i].height;
+        vk_viewports[i].minDepth = pViewports[i].minDepth;
+        vk_viewports[i].maxDepth = pViewports[i].maxDepth;
+    }
 
-                uint32_t geometryArrayOffset = 0u;
-                VkAccelerationStructureGeometryKHR vk_geometries[MaxGeometryPerBuildInfoCount * MaxBuildInfoCount] = {};
+    const auto* vk = static_cast<const CVulkanLogicalDevice*>(getOriginDevice())->getFunctionTable();
+    vk->vk.vkCmdSetViewport(m_cmdbuf, firstViewport, viewportCount, vk_viewports);
+    return true;
+}
 
-                IGPUAccelerationStructure::DeviceBuildGeometryInfo* infos = pInfos.begin();
-                for(uint32_t i = 0; i < infoCount; ++i)
-                {
-                    uint32_t geomCount = infos[i].geometries.size();
-
-                    assert(geomCount > 0);
-                    assert(geomCount <= MaxGeometryPerBuildInfoCount);
-
-                    vk_buildGeomsInfos[i] = CVulkanAccelerationStructure::getVkASBuildGeomInfoFromBuildGeomInfo(vk_device, vk, infos[i], &vk_geometries[geometryArrayOffset]);
-                    geometryArrayOffset += geomCount; 
-                    
-                    // Add Refs of BuildInfo to CmdPool
-                    auto tmp = getBuildGeometryInfoReferences(infos[i]);
-                    CVulkanCommandPool* vulkanCommandPool = IBackendObject::compatibility_cast<CVulkanCommandPool*>(m_cmdpool.get(), this);
-                    vulkanCommandPool->emplace_n(m_argListTail, tmp.data(), tmp.data() + tmp.size());
-                }
-                
-                static_assert(sizeof(IGPUAccelerationStructure::BuildRangeInfo) == sizeof(VkAccelerationStructureBuildRangeInfoKHR));
-                auto buildRangeInfos = reinterpret_cast<const VkAccelerationStructureBuildRangeInfoKHR* const*>(ppBuildRangeInfos);
-                vk->vk.vkCmdBuildAccelerationStructuresKHR(m_cmdbuf, infoCount, vk_buildGeomsInfos, buildRangeInfos);
-                ret = true;
-            }
-        }
-        return ret;
+bool CVulkanCommandBuffer::copyBuffer_impl(const buffer_t* srcBuffer, buffer_t* dstBuffer, uint32_t regionCount, const asset::SBufferCopy* pRegions)
+{
+    VkBuffer vk_srcBuffer = IBackendObject::compatibility_cast<const CVulkanBuffer*>(srcBuffer, this)->getInternalObject();
+    VkBuffer vk_dstBuffer = IBackendObject::compatibility_cast<const CVulkanBuffer*>(dstBuffer, this)->getInternalObject();
+
+    constexpr uint32_t MAX_BUFFER_COPY_REGION_COUNT = 681u;
+    VkBufferCopy vk_bufferCopyRegions[MAX_BUFFER_COPY_REGION_COUNT];
```

Comment:
missing an assert that `regionCount` is less than the max

Reply:
Resolved.